db/db.go

```diff
@@ -152,9 +153,9 @@ func newDB(n net.Net, creds thread.Credentials, config *Config) (*DB, error) {
 		}
 	}
 
-	adapter := newSingleThreadAdapter(d, n, creds)
-	d.adapter = adapter
-	adapter.Start()
+	connector := n.ConnectDB(d, id)
+	d.connector = connector
+	connector.Start()
```

Comment:
Thought: Does `Start()` method stil makes sense now that things are auto-started?

Reply:
Nope, removing.